platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/TeachingSynchronizer.java

```diff
@@ -101,7 +104,8 @@ public TeachingSynchronizer(
             final MerkleDataInputStream in,
             final MerkleDataOutputStream out,
             final MerkleNode root,
-            final Runnable breakConnection) {
+            final Runnable breakConnection,
+            final Supplier<Boolean> requestToStopTeaching) {
```

Comment:
nit, you can use the type `java.util.function.BooleanSupplier` to get a primitive `boolean` instead of `Boolean

Reply:
Agree. Let me change it to `BooleanSupplier`